### DISCUSSION TITLE
Alternative backend APIs

### DISCUSSION THREAD

**Discussion by erlend-sh** [likeandscribe/frontpage#169] [Ideas]
_From @tom-sherman :_

> we spoke before about cloning the frontpage repo and working on it locally, i didn't have a good answer before but over the last couple of days i've spun up a docker compose stack for local use 
> 
> with this there's no hard dependency on bluesky or a cloud db, it all runs on your machine
> 
> https://github.com/likeandscribe/frontpage/tree/main/packages/frontpage/local-infra

This seems to imply that frontpage is in essence backend-agnostic, am I understanding that correctly?

Could this frontpage-ui be extended to also support Lemmy? (That’d make it a 🧃[juicy client](https://blog.erlend.sh/juicy-clients)!)

It’s more of a general question about the feasibility of supporting Lemmy API as an alternative backend, and weaving frontpage-Lemmy posts in alongside  frontpage-atproto posts.

If this is possible, there are three essential backend APIs I wanna intermingle together in such a UI:
- [atproto API](https://atproto.com/guides/applications) (ATmosphere-net)
- [Lemmy API](https://join-lemmy.org/docs/contributors/04-api.html) (which by extension connects to the ActivityPub-derived threadiverse-net)
- [HackerNews API](https://github.com/HackerNews/API) (hn-net)

**Comment by tom-sherman** [comment]
Hmm, that isn't what I intended by my message but I can see the use case here.

A UI that shows posts from different networks is probably out of scope of Frontpage as we're trying to be an atproto app (a la Bluesky), our features are going to become more tightly coupled to the protocol over time (starting with communities). Not to discount bridging those posts to Frontpage as bridgy fed does for the Bluesky lexicon right now.

I agree that such a UI is a good idea though and it could be built on top of a future Frontpage API (when we get around to implementing it 😅).
